consensus/phases/data_serializers.go

```diff
@@ -0,0 +1,677 @@
+/*
+ *    Copyright 2018 Insolar
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package phases
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+
+	"github.com/insolar/insolar/network/transport"
+	"github.com/pkg/errors"
+)
+
+var defaultByteOrder = binary.BigEndian
+
+type Serializer interface {
+	Serialize() ([]byte, error)
+	Deserialize(data io.Reader) error
+}
+
+// ----------------------------------PHASE 1--------------------------------
+
+// routInfoMasks masks
+const (
+	// take low bit
+	hasRoutingMask = 0x1
+
+	packetTypeMask = 0xf0
+	subtypeMask    = 0xe
+)
+
+func (ph *PacketHeader) parseRouteInfo(routInfo uint8) {
+	ph.PacketT = PacketType((routInfo & packetTypeMask) >> 4)
+	ph.SubType = (routInfo & subtypeMask) >> 1
+	ph.HasRouting = (routInfo & hasRoutingMask) == 1
+}
+
+func (ph *PacketHeader) compactRouteInfo() uint8 {
+	var result uint8
+	result |= uint8(ph.PacketT) << 4
+	result |= ph.SubType << 1
+
+	if ph.HasRouting {
+		result |= hasRoutingMask
+	}
+
+	return result
+}
+
+// PulseAndCustomFlags masks
+const (
+	// take bit before high bit
+	f00Mask = 0x40000000
+
+	// take high bit
+	f01Mask   = 0x80000000
+	pulseMask = 0x3fffffff
+)
+
+func (ph *PacketHeader) parsePulseAndCustomFlags(pulseAndCustomFlags uint32) {
+	ph.F01 = (pulseAndCustomFlags >> 31) == 1
+	ph.F00 = ((pulseAndCustomFlags & f00Mask) >> 30) == 1
```

Comment:
define constants for magic numbers 30 and 31

Reply:
done